README.md

```diff
@@ -32,3 +32,382 @@ cd loopback4-example-getting-started && npm install
+#### src/models/todo.model.ts
+```ts
+import {Entity, property, model} from '@loopback/repository';
+import {SchemaObject} from '@loopback/openapi-spec';
+
+@model()
+export class Todo extends Entity {
+  @property({
+    type: 'number',
+    id: true
+  })
+  id?: number;
+
+  @property({
+    type: 'string',
+    required: true
+  })
+  title: string;
+
+  @property({
+    type: 'string'
+  })
+  desc?: string;
+
+  @property({
+    type: 'boolean'
+  })
+  isComplete: boolean;
+
+  getId() {
+    return this.id;
+  }
+}
+
+export const TodoSchema: SchemaObject = {
+  title: 'todoItem',
+  properties: {
+    id: {
+      type: 'number',
+      description: 'ID number of the Todo entry.'
+    },
+    title: {
+      type: 'string',
+      description: 'Title of the Todo entry.'
+    },
+    desc: {
+      type: 'number',
+      description: 'ID number of the Todo entry.'
+    },
+    isComplete: {
+      type: 'boolean',
+      description: 'Whether or not the Todo entry is complete.'
+    }
+  },
+  required: ['title'],
+};
+```
+
+### Building a Datasource
+Before we can begin constructing controllers and repositories for our
+application, we need to define our datasource.
+
+Create a new folder in the root directory of the project called `config`,
+and then inside that folder, create a `datasources.json` file. For now, we'll
+be using the memory connector provided with the Juggler.
+
+#### config/datasources.json
+```json
+{
+  "name": "ds",
+  "connector": "memory"
+}
+```
+
+Create another folder called `datasources` in the `src` directory, and inside
+that folder, create a new file called `db.datasource.ts`.
+
+#### src/datasources/db.datasource.ts
+
+```ts
+import * as path from 'path';
+import * as fs from 'fs';
+import { DataSourceConstructor, juggler } from '@loopback/repository';
+
+const dsConfigPath = path.resolve('config', 'datasources.json');
+const config = require(dsConfigPath);
+export const db = new DataSourceConstructor(config);
+```
+
+This will give us a strongly-typed datasource export that we can work with to
+construct our TodoRepository definition.
+
+### Create your repository
+Create another folder in `src` called `repositories` and inside of that folder,
+create two files:
+- `index.ts` (our export helper)
+- `todo.repository.ts`
+
+Our TodoRepository will contain a small base class that uses the
+`DefaultCrudRepository` class from `@loopback/repository` and will define the
+model type we're working with, as well as its ID type. We'll also inject our
+datasource so that this repository can connect to it when executing data
+operations.
+
+#### src/repositories/todo.repository.ts
+```ts
+import { DefaultCrudRepository, DataSourceType } from '@loopback/repository';
+import { Todo } from '../models';
+import { inject } from '@loopback/core';
+
+export class TodoRepository extends DefaultCrudRepository<
+  Todo,
+  typeof Todo.prototype.id
+> {
+  constructor(@inject('datasource') protected datasource: DataSourceType) {
```

Comment:
nitpick, maybe refer to [application.ts](https://github.com/strongloop/loopback4-example-getting-started/pull/11/files#diff-04c6e90faac2675aa89e2176d2eec7d8R395) so readers know where is the `datasource` bound to. Just feel a little bit weird to see the injection code first...or maybe that's the way how dependency injection works?